Sources/Amplitude/Amplitude.m

```diff
@@ -284,6 +272,49 @@ - (instancetype)initWithInstanceName:(NSString *)instanceName {
     return self;
 }
 
+- (void)migrateToFileStorage {
+     NSString *databaseDirectory = [AMPUtils platformDataDirectory];
+     NSString *databasePath = [databaseDirectory stringByAppendingPathComponent:@"com.amplitude.database"];
+    if (![self->_instanceName isEqualToString:kAMPDefaultInstance]) {
+        databasePath = [NSString stringWithFormat:@"%@_%@", databasePath, self.instanceName];
+    }
+    NSString *fileStoragePath =  [AMPStorage getAppStorageAmpDir:self.instanceName];
+
+    BOOL firstMigration = ![[NSFileManager defaultManager] fileExistsAtPath:fileStoragePath isDirectory: true];
+    BOOL needDBMigrate =  ([[NSFileManager defaultManager] fileExistsAtPath:databasePath] && firstMigration) ? true : false;
+    if (!needDBMigrate) return;
+    int eventCount = [self.dbHelper getEventCount];
+    if (eventCount > 0) {
+        NSArray *events = [self.dbHelper getEvents: -1 limit: -1];
+        NSDictionary *eventsDict = [self mergeEventsAndIdentifys:events identifys:nil numEvents:eventCount];
+        long long maxEventId = [[eventsDict objectForKey:MAX_EVENT_ID] longLongValue];
+        for(NSString *event in events) {
+            [AMPStorage storeEvent:event instanceName:self.instanceName];
```

Comment:
So when would we actually load this event? After we save it to the file, in logEvent it would not read from the file right? Per @dantetam's last pr, it will only load the file when it gets to foreground? Would we just happen to truncate or remove the file but not sending the old events?

Reply:
I think that should not happen. 
1. in the storeEvent, we have logic that if we already create the txt file we only append events at the end of the file. 
2. we only remove the files after we load the event to the server and get success response.